speaker/layer2_controller.go

```diff
@@ -133,9 +142,20 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	return "notOwner"
 }
 
+func getInterfaces(localNode string, l2Advertisements []*config.L2Advertisement) []string {
+	ifs := sets.NewString()
```

Comment:
This won't handle the case where you have one advertisement with no interface (and thus, collecting all of them) and one other advertisement with a selector.
The desired result is to advertise all of them, but we'll advertise to a subset only.

Reply:
I has handle this case when announcer，if len(interfaces)==0，it won't filter interface.

Reply:
Yes, but the problem is, if you have two l2 advertisements, one with some interfaces and another one with no interfaces, the semantic there is : advertise to all (because you have one l2 advertisement with no interfaces) but your if won't trigger because len(interfaces) is not 0

Reply:
yes, I'm ill-considered.